relay/redis.go

```diff
@@ -0,0 +1,68 @@
+package relay
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/batchcorp/schemas/build/go/events/records"
+	"github.com/batchcorp/schemas/build/go/services"
+	"google.golang.org/grpc"
+
+	"github.com/batchcorp/plumber/backends/redis/types"
+)
+
+// handleRedis sends a Redis relay message to the GRPC server
+func (r *Relay) handleRedis(ctx context.Context, conn *grpc.ClientConn, messages []interface{}) error {
+	sinkRecords, err := r.convertMessagesToRedisSinkRecords(messages)
+	if err != nil {
+		return fmt.Errorf("unable to convert messages to redis sink records: %s", err)
+	}
+
+	client := services.NewGRPCCollectorClient(conn)
+
+	return r.CallWithRetry(ctx, "AddRedisRecord", func(ctx context.Context) error {
+		_, err := client.AddRedisRecord(ctx, &services.RedisRecordRequest{
+			Records: sinkRecords,
+		}, grpc.MaxCallRecvMsgSize(MaxGRPCMessageSize))
+		return err
+	})
+}
+
+// validateRedisRelayMessage ensures all necessary values are present for a Redis relay message
+func (r *Relay) validateRedisRelayMessage(msg *types.RelayMessage) error {
+	if msg == nil {
+		return errMissingMessage
+	}
+
+	if msg.Value == nil {
+		return errMissingMessageValue
+	}
+
+	return nil
+}
+
+// convertRedisMessageToProtobufRecord creates a records.RedisSinkRecord from a redis.Message which can then
+// be sent to the GRPC server
+func (r *Relay) convertMessagesToRedisSinkRecords(messages []interface{}) ([]*records.RedisRecord, error) {
+	sinkRecords := make([]*records.RedisRecord, 0)
+
+	for i, v := range messages {
+		relayMessage, ok := v.(*types.RelayMessage)
+		if !ok {
+			return nil, fmt.Errorf("unable to type assert incoming message as RelayMessage (index: %d)", i)
```

Comment:
Should these two returns just be logs and continue? Otherwise it would reject the whole batch

Reply:
Yeah definitely - missed it. Thanks!

Reply:
Going to skip for now - this should include a better mechanism that reports how much data was actually sent. Just doing `log.Errorf(..)` is not enough (especially if running plumber in docker).

Let's address in a future PR.